patches/net/minecraft/server/level/ServerPlayerGameMode.java.patch

```diff
@@ -58,18 +58,22 @@
                      ItemStack itemstack = this.player.getMainHandItem();
                      ItemStack itemstack1 = itemstack.copy();
 -                    boolean flag = this.player.hasCorrectToolForDrops(blockstate);
-+                    boolean flag1 = blockstate.canHarvestBlock(this.level, p_9281_, this.player); // previously player.hasCorrectToolForDrops(blockstate)
-                     itemstack.mineBlock(this.level, blockstate, p_9281_, this.player);
+-                    itemstack.mineBlock(this.level, blockstate, p_9281_, this.player);
+-                    if (flag1 && flag) {
+-                        block.playerDestroy(this.level, this.player, p_9281_, blockstate, blockentity, itemstack1);
++                    BlockState removedBlockState = blockstate.getBlock().playerWillDestroy(this.level, p_9281_, blockstate, this.player);
```

Comment:
Should probably move this call back up to where it was originally now that the methods are decoupled, just after 
the `} else {` line.  That would mean we don't need to reintroduce the new local and update all the usage sites.

`BlockState blockstate = block.playerWillDestroy(this.level, p_9281_, blockstate1, this.player);`

Reply:
good call, moving it now